lib/adapters/idb/allDocs.js

```diff
@@ -59,7 +59,7 @@ function handleKeyRangeError(api, opts, err, callback) {
 function idbAllDocs(opts, api, idb, callback) {
 
   function allDocsQuery(opts, callback) {
-    var start = 'startkey' in opts ? opts.startkey : false;
+    var start = 'startkey' || 'start_key' in opts ? opts.startkey : false;
```

Comment:
So theres 2 issues here, the first one is mostly because as mentioned above I didnt explain the bug properly. 

first is when the use gives us `start_key`, we want to treat it exactly as if they gave us `startkey`. The logic here wont quite do that but the code will be exactly the same as last time, ie `if (opts.start_key) { opts.startkey = opts.start_key; }`

The second is due to how our code is setup which is pretty confusing, you have fixed this in 4 places (http, leveldb, websql and idb), http is special for now but for the rest you can fix them all in the same place. 

We call the function @ https://github.com/pouchdb/pouchdb/blob/master/lib/adapter.js#L645 which will in turn call these implementations, so any common logic especially with the parameters can go in there once and be fixed for all 3 adapters (like how we set a default value for skip @ https://github.com/pouchdb/pouchdb/blob/master/lib/adapter.js#L652)

So if you do what you did for the http adapter and put in that function just around where we do the `opts.skip` stuff, it will only need fixed in the one place.